types/src/system/auction/providers.rs

```diff
@@ -93,6 +93,16 @@ pub trait MintProvider {
     /// Reduce total supply by `amount`. Returns unit on success, otherwise
     /// an error.
     fn reduce_total_supply(&mut self, amount: U512) -> Result<(), Error>;
+
+    /// Transfers token from a given source URef to a bonding purse.
+    /// Returns the number of tokens currently delegated to a given validator.
+    fn handle_delegation(
```

Comment:
This method should be a a free function in auction's `detail.rs` as MintProvider trait is meant to be a bridge between Auction contract and Mint system contract